src/libAtomVM/nifs.c

```diff
@@ -2466,3 +2478,161 @@ static term nif_console_print(Context *ctx, int argc, term argv[])
     }
     return OK_ATOM;
 }
+
+static char b64_table[64] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+
+// per https://tools.ietf.org/rfc/rfc4648.txt
+
+static term nif_atomvm_b64_encode(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+    term src = argv[0];
+
+    VALIDATE_VALUE(src, term_is_binary);
+
+    size_t src_size = term_binary_size(src);
+    if (src_size == 0) {
+        return src;
+    }
+    uint8_t *src_buf = malloc(src_size);
+    if (IS_NULL_PTR(src_buf)) {
+        RAISE_ERROR(OUT_OF_MEMORY_ATOM);
+    }
+    memcpy(src_buf, term_binary_data(src), src_size);
```

Comment:
we don't need to copy src to a temporary buffer, it is safe to use the pointer returned by term_binary_data(src).

Reply:
Makes sense (see below)